src/components/DialogHeader/DialogHeader.tsx

```diff
@@ -0,0 +1,70 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { ReactElement } from 'react';
+import { merge } from '@utilities/merge';
+import { IconCross } from '@foundation/Icon/Generated';
+import { Text } from '@typography/Text';
+
+export type DialogHeaderProps = {
+    title: string;
+    size?: DialogHeaderSize;
+    collapseBottom?: boolean;
+    onClose?: () => void;
+    icon?: ReactElement;
+    badge?: ReactElement;
```

Comment:
Maybe we can set badge and icon props to be `BadgeProps`, `IconProps`. Being a `ReactElement` means we can pass something different than a `Badge` and `Icon` for them...

Reply:
Good point, I set a type to the ReactElement. Like that we don't have Icon: ReactElement and Badge: Badgeprops. 
A react Element needs to be passed for both properties. It's more consistent.